j5/backends/hardware/j5/arduino.py

```diff
@@ -96,26 +107,52 @@ def __init__(
     @property
     @abstractmethod
     def firmware_version(self) -> Optional[str]:
-        """The firmware version of the board."""
+        """
+        The firmware version reported by the board.
+
+        :returns: firmware version reported by the board, if any.
+        """
         raise NotImplementedError  # pragma: nocover
 
     @abstractmethod
     def _update_digital_pin(self, identifier: int) -> None:
-        """Write the stored value of a digital pin to the Arduino."""
+        """
+        Write the stored value of a digital pin to the Arduino.
+
+        Reads the state out of self._digital_pins.
```

Comment:
Yes this is correct. This is the method on the abstract Arduino implementation (base class for SBArduino and Ruggeduino), rather than on the `GPIOPin` interface.

The method you are thinking of is `GPIOPinInterface.set_gpio_pin_mode`, which is implemented using `_update_digital_pin`.